README.md
## modal_drawer_handle

11/17/2018 NOTE: This package is being folded into my new widget package 'groovin_widgets'. This package will not be updated further.

modal_drawer_handle is A Flutter package that allows you to add a highly customizable drawer handle widget to your modalBottomSheets.

Sample images: https://imgur.com/gallery/zf6NriP

YouTube video walkthrough: https://www.youtube.com/watch?v=VF-zR9ougi8&feature=youtu.be

Every aspect of the drawer handle can be customized:
- color
- height
- width
- the MainAxisAlignment of the Row it sits in
- the BorderRadius of all four corners

The recommended use of this widget is within the showModalBottomSheet builder (or the showRoundedModalBottomSheet builder, if you prefer the package by Gildaswise like I do).

The ideal use in this case is to return a Container that returns a Column, and the ModalDrawerHandle widget should be the first widget in the Column. I personally wrap the ModalDrawerHandle with a Padding that has a const EdgeInsets.all(8.0). Below is an example of this:

```
showRoundedModalBottomSheet(
  context: context,
  builder: (builder) {
    return Container(
	  child: Column(
	    mainAxisSize: MainAxisSize.min,
	    children: <Widget>[
		  Padding(
		    padding: const EdgeInsets.all(8.0),
		    child: ModalDrawerHandle(),
		  ),
	    ],
	  ),
    );
  },
);
```

By default, the ModalDrawerHandle is initialized with the following properties:
```
  ModalDrawerHandle({
    this.handleRowAlignment = MainAxisAlignment.center,
    this.handleHeight = 5.0,
    this.handleWidth = 25.0,
    Color handleColor,
    this.handleBorderRadius = const BorderRadius.all(Radius.circular(10.0))
  }):
     this.handleColor = handleColor ?? Colors.grey[300];
```

As such, developers can pass in any or all of their own custom values to the constructor as they choose. 

## Getting Started With Flutter

For help getting started with Flutter, view our online
[documentation](https://flutter.io/).

CHANGELOG.md
## [1.0.4] - Released 11/17/2018

* Updated README

## [1.0.3] - Released 11/10/2018

* Updated README with image and video links

## [1.0.2] - Released 11/9/2018

* Ran dartfmt

## [1.0.1] - Released 11/9/2018

* Fixed changelog, docs, and version.

## [1.0.0] - Released 11/9/2018

* Initial release.


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:groovin_material_icons/groovin_material_icons.dart';
import 'package:rounded_modal/rounded_modal.dart';
import 'package:modal_drawer_handle/modal_drawer_handle.dart';
import 'package:url_launcher/url_launcher.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      title: 'ModalDrawerHandle demo',
      theme: ThemeData(
        primaryColor: Colors.indigo,
        accentColor: Colors.indigo,
      ),
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  Color randomHandleColor = Colors.indigo;

  Color randomColor() {
    final rColor = (List<Color>()
          ..addAll(Colors.primaries)
          ..shuffle())
        .first;
    return rColor;
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        elevation: 0.0,
        centerTitle: true,
        backgroundColor: Theme.of(context).canvasColor,
        title: Text(
          "ModalDrawerHandle Demo",
          style: TextStyle(
            color: Theme.of(context).brightness == Brightness.light
              ? Colors.black
              : Colors.white),
        ),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            RaisedButton(
              child: Text(
                "Change handle color (random)",
                style: TextStyle(
                  color: Colors.white,
                ),
              ),
              color: Colors.indigoAccent,
              onPressed: () {
                setState(() {
                  randomHandleColor = randomColor();
                });
              },
            ),
          ],
        ),
      ),
      floatingActionButtonLocation: FloatingActionButtonLocation.centerFloat,
      floatingActionButton: FloatingActionButton.extended(
        icon: Icon(Icons.keyboard_arrow_up),
        label: Text("Show drawer"),
        backgroundColor: randomHandleColor,
        onPressed: () {
          showRoundedModalBottomSheet(
            context: context,
            builder: (builder) {
              return Container(
                child: Column(
                  mainAxisSize: MainAxisSize.min,
                  children: <Widget>[
                    Padding(
                      padding: const EdgeInsets.all(8.0),
                      child: ModalDrawerHandle(
                        handleColor: randomHandleColor,
                      ),
                    ),
                    ListTile(
                      leading: Icon(Icons.info_outline),
                      title: Text("Package developed by GroovinChip"),
                      subtitle: Text("Version 1.0.0"),
                    ),
                    Material(
                      child: ListTile(
                        leading: Icon(GroovinMaterialIcons.github_circle),
                        title: Text("View source code"),
                        onTap: () {
                          launch("https://github.com/GroovinChip/modal_drawer_handle");
                        },
                      ),
                    ),
                    Material(
                      child: ListTile(
                        leading: Icon(GroovinMaterialIcons.twitter),
                        title: Text("Tweet @GroovinChipDev"),
                        onTap: () {
                          launch("https://twitter.com/GroovinChipDev/");
                        },
                      ),
                    ),
                  ],
                ),
              );
            },
          );
        },
      ),
    );
  }
}


lib/modal_drawer_handle.dart
library modal_drawer_handle;

import 'package:flutter/material.dart';

/// This widget represents a customizable 'handle' that you can add to your
/// modalBottomSheets
class ModalDrawerHandle extends StatelessWidget {
  /// Represents the MainAxisAlignment of the Row that contains the handle.
  /// The default value is MainAxisAlignment.center
  final MainAxisAlignment handleRowAlignment;

  /// Represents the height of the handle. The default value is 5.0
  final double handleHeight;

  /// Represents the height of the handle. The default value is 25.0
  final double handleWidth;

  /// Represents the color of the handle.
  /// The default value is #EEEEEE
  final Color handleColor;

  /// Represents the BorderRadius (rounding the corners of the handle).
  /// The default value is const BorderRadius.all(Radius.circular(10.0)).
  /// Note that when using a custom BorderRadius, you need to pass in
  /// a Radius, not a double.
  final BorderRadius handleBorderRadius;

  /// ModalDrawerHandle constructor.
  /// The widget is initialized with the default values.
  ModalDrawerHandle(
      {this.handleRowAlignment = MainAxisAlignment.center,
      this.handleHeight = 5.0,
      this.handleWidth = 25.0,
      Color handleColor,
      this.handleBorderRadius = const BorderRadius.all(
        Radius.circular(10.0),
      )})
      : this.handleColor = handleColor ?? Colors.grey[300];

  // ModalDrawerHandle Widget builder
  @override
  Widget build(BuildContext context) {
    return Row(
      mainAxisAlignment: this.handleRowAlignment,
      children: <Widget>[
        Container(
          height: this.handleHeight,
          width: this.handleWidth,
          decoration: BoxDecoration(
            color: this.handleColor,
            borderRadius: this.handleBorderRadius,
          ),
        ),
      ],
    );
  }
}